core/include/detray/propagator/propagator.hpp

```diff
@@ -71,60 +76,50 @@ struct propagator {
         DETRAY_HOST_DEVICE state(
             const free_track_parameters_type &t_in, const detector_type &det,
             vector_type<intersection_type> &&candidates = {})
-            : _stepping(t_in),
-              _navigation(det, std::move(candidates)),
-              m_param_type(parameter_type::e_free) {}
+            : m_param_type(parameter_type::e_free),
+              _stepping(t_in),
+              _navigation(det, std::move(candidates)) {}
 
         template <typename field_t>
         DETRAY_HOST_DEVICE state(
             const free_track_parameters_type &t_in,
             const field_t &magnetic_field, const detector_type &det,
             vector_type<intersection_type> &&candidates = {})
-            : _stepping(t_in, magnetic_field),
-              _navigation(det, std::move(candidates)),
-              m_param_type(parameter_type::e_free) {}
+            : m_param_type(parameter_type::e_free),
+              _stepping(t_in, magnetic_field),
+              _navigation(det, std::move(candidates)) {}
 
         /// Construct the propagation state with bound parameter
         DETRAY_HOST_DEVICE state(
             const bound_track_parameters_type &param, const detector_type &det,
             vector_type<intersection_type> &&candidates = {})
-            : _stepping(param, det),
-              _navigation(det, std::move(candidates)),
-              m_param_type(parameter_type::e_bound) {}
+            : m_param_type(parameter_type::e_bound),
+              _stepping(param, det),
+              _navigation(det, std::move(candidates)) {}
 
         /// Construct the propagation state with bound parameter
         template <typename field_t>
         DETRAY_HOST_DEVICE state(
             const bound_track_parameters_type &param,
             const field_t &magnetic_field, const detector_type &det,
             vector_type<intersection_type> &&candidates = {})
-            : _stepping(param, magnetic_field, det),
-              _navigation(det, std::move(candidates)),
-              m_param_type(parameter_type::e_bound) {}
+            : m_param_type(parameter_type::e_bound),
+              _stepping(param, magnetic_field, det),
+              _navigation(det, std::move(candidates)) {}
 
         DETRAY_HOST_DEVICE
         parameter_type param_type() const { return m_param_type; }
 
         DETRAY_HOST_DEVICE
         void set_param_type(const parameter_type t) { m_param_type = t; }
 
-        DETRAY_HOST_DEVICE
```

Comment:
Is this necessary?

Reply:
I would like to have the navigator configuration in one place